libs/cardano-ledger-test/src/Test/Cardano/Ledger/Examples/ConwayFeatures.hs

```diff
@@ -12,6 +13,7 @@
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE TypeOperators #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
```

Comment:
We should refrain from this I think. It would be hard to debug these tests.

Reply:
I don't think it's as bad if we evaluate all the `let` statements strictly, then it sort of reads like python or C code :smiley: . It can be hard to give each variable a unique name and when something gets added in the middle we have to re-enumerate everything.

Reply:
Sorry, @Soupstraw but we only disable compiler warning when we really have to. So, I agree with Aniket. Could you please remove this GHC flag.

Reply:
> I don't think it's as bad if we evaluate all the let statements strictly, then it sort of reads like python or C code 😃 . It can be hard to give each variable a unique name and when something gets added in the middle we have to re-enumerate everything.

Well, I would agree with you, but let me explain what I thought about this: all conventional haskell programmers are used to making all sorts of assumptions in their heads due to the default semantics of the language. When one changes the default semantics of the language, the indicator that some part of the code is to be treated specially is best left right next to the site where the semantics is changed, so other programmers can notice it. Here we are proposing to suppress name-shadowing warnings file-wide, and keeping the indicator at the top of the file. Any engineer reading the code 3 months from now, including you, has a high (read total) probability of being confused by this. So if and when we are sure that something will always be a problem, we should put it in bad-practice category. Bad practice or "code-smells" are typically exactly like this: they feel like a cool short-cut but bite us back in the long term.